command-line/sim_interface.py

```diff
@@ -0,0 +1,171 @@
+        self.data_generator = DataGeneration
+
+
+    def configure_device(self, device : VCUSimulation):
+        """
+        Configure the device from a json file. Currently unimplemented
+        but could be useful for setting baudrate, or the OS ports
+        """
+        json_config: dict = self._parse_json()
+        self.set_metadata(device, **json_config)
+
+    def _parse_json(self) -> Union[dict, None]:
+        """
+        Parse the JSON containing the VCU specs
+        """
+       
+        json_file_path = 'configs/device_config.json'
+
+        # Check if the directory exists
+        json_directory = os.path.dirname(json_file_path)
+
+        if not os.path.exists(json_directory):
+            return dict()
+        else:
+            # Load the JSON file
+            with open(json_file_path, 'r') as json_file:
+                data = json.load(json_file)
+
+                #return config data within
+                return data.get('VCU', {})
+        
+
+    def set_metadata(self, device, **config):
+        """
+        Configure the simulation peripherals based on the config
+        """
+        self.vcu_writer = device(**config)
+
+    def begin(self):
+        """
+        Initialization point of the simulation interface. Executes the simulation, retrieves the results
+        and finally uploads the data to the Google Drive.
+        """
+        simulation_res : bool = self.simulation.begin()
+        if (simulation_res):
+            write_res : dict[Union[VCU_Pedal, str], Union[list[int], list[EventData], list[StateData]]] = self._write_data()
+            self.data_generator.write_to_csv(write_res)
+            self.vcu_gdrive_interface.upload_data()
+            exit()
+        else:
+            #Handle 
+            raise Exception("Problem with building simulation")
+
+    def _write_data(self) -> dict[Union[VCU_Pedal, str], Union[list[int], list[EventData], list[StateData]]]:
+        """
+        Write the generated simulation to the VCU firmware. Retrieve the responses for the events and state changes. 
+        """
+        
+        sim_model : dict[VCU_Pedals, list[float]] = self.simulation.plotted_points.copy()
+        #all the lenghts of the sim model are the same  
+        sim_len = len(sim_model[VCU_Pedals.APPS1])
+
+        
+        res_data = { StateData.__name__ : [], EventData.__name__: [], "Raw Response": []}
+
+        time_data = {"Time (ms)" : []}
+
+        start_time = time.time()
+
+        #retrieve data from response
+        for sim_num in range(sim_len):
+
+            apps1: float = sim_model[VCU_Pedals.APPS1][sim_num]
+            apps2: float = sim_model[VCU_Pedals.APPS2][sim_num]
+            bse: float = sim_model[VCU_Pedals.BSE][sim_num]
+
+            end_time = time.time()
+
+            
+            # response_vcu: ResponseVCU = ResponseVCU(self.vcu_writer.write(int(apps1), int(apps2), int(bse)))
```

Comment:
Are these lines for the user to uncomment or extra code that needs cleanup?

Reply:
Fixed, they art part of the framework execution and should have been uncommented. I accidentally commited these changes while testing the formatting

bfcd917f06f4cbad3e850ea23e66865096742888